daemon/list.go

```diff
@@ -26,21 +46,74 @@ type ContainersConfig struct {
 	Filters string
 }
 
+// listContext is the daemon generated filtering to iterate over containers.
+// This is created based on the user specification.
+type listContext struct {
+	idx             int
+	names           map[string][]string
+	filters         filters.Args
+	exitAllowed     []int
+	beforeContainer *Container
+	sinceContainer  *Container
+	*ContainersConfig
+}
+
+// Containers returns the list of containers to show given the user's filtering.
 func (daemon *Daemon) Containers(config *ContainersConfig) ([]*types.Container, error) {
-	var (
-		foundBefore bool
-		displayed   int
-		all         = config.All
-		n           = config.Limit
-		psFilters   filters.Args
-		filtExited  []int
-	)
+	return daemon.reduceContainers(config, daemon.transformContainer)
```

Comment:
yep, this is the whole point of the refactor. It will be easier to version the api output if we only have to decide which transformation to use.